src/platform/windows.c

```diff
@@ -1126,7 +1117,7 @@ static bool windows_setup_file_association(
 	}
 	// [hRootKey\OpenRCT2.ext\DefaultIcon]
 	wchar_t szIconW[MAX_PATH];
-	wsprintfW(szIconW, L"\"%s\",%d", dllPathW, iconIndex);
+	StringCchPrintfW(szIconW, MAX_PATH, L"\"%s\",%d", dllPathW, iconIndex);
```

Comment:
As long as the function always adds the null terminator, even if the string is longer the the buffer size then its fine.